### PR TITLE
Treat blocks with "return" as "noreturn"

### LINKED ISSUE #11686: `pub fn main() void { { return; } return; }` evades unreachable code error
### Zig Version

0.10.0-dev.2315+dd6ac9a22

### Steps to Reproduce

```zig
pub fn main() void {
    {
        return;
    }
    return;
}
```

### Expected Behavior

```
main.zig:5:5: error: unreachable code
    return;
    ^
main.zig:3:5: note: control flow is diverted here
    return;
    ^
```

### Actual Behavior

A successful compilation.

### PR DESCRIPTION
**Pull request by gpanders**
Block statements that end with "break" should not be considered
"noreturn" for the enclosing scope, but other "noreturn" instructions
(return, panic, compile error, etc.) should be. This differentiation
necessitates handling "break" differently from the other "noreturn"
instructions when inside a block statement.

Closes #11686.

### CODE REVIEW

File: src/Zir.zig [hunk 1]
```diff
@@ -1250,6 +1250,271 @@ pub const Inst = struct {
...
+                .compile_error,
+                .ret_node,
+                .ret_load,
+                .ret_tok,
+                .ret_err_value,
+                .@"unreachable",
+                .repeat,
+                .repeat_inline,
+                .panic,
+                => false,
```
Comment (Vexu) [inline src/Zir.zig:1510]: I think it's pretty safe to replace these 200 lines with an `else => false`.

Comment (gpanders) [reply src/Zir.zig:1510]: Sounds good. I listed them explicitly to copy `endsWithNoReturn()`, as I wasn't sure if that was done for a specific reason.

**Comment by andrewrk** [comment]
I think the patch should fully solve the issue in this case; this halfway solution is doing more harm than good. `endsWithBreak()` is a footgun.

> This didn't apply to labeled blocks.

Not sure what you mean by "this" - did you mean the logic `if (gz.endsWithNoReturn() and !gz.endsWithBreak()) {` was in `blockExpr`? That's true in my example above as well. The break target is labeled, but the containing block is unlabeled.

**Comment by gpanders** [comment]
@andrewrk I didn't think of the nested block labels. I'll circle back to this and give it a closer look.

**Comment by andrewrk** [comment]
Thanks @gpanders! I always feel bad doing a revert, please don't take it personally. Just another part of the development process.

**Pull request merged**